File: app/src/main/java/me/echeung/moemoekyun/socket/response/UpdateResponse.java
package me.echeung.moemoekyun.socket.response;

import java.util.List;

import lombok.Getter;
import me.echeung.moemoekyun.model.Event;
import me.echeung.moemoekyun.model.Song;
import me.echeung.moemoekyun.model.User;

@Getter
public class UpdateResponse extends BaseResponse {
    private String t;
    private Details d;

    @Getter
    public class Details {
        private Song song;
        private String startTime;
        private List<Song> lastPlayed;
        private Queue queue;
        private int listeners;
        private User requester;
        private Event event;
    }

    @Getter
    public class Queue {
        private int inQueue;
        private int inQueueByUser;
        private int inQueueBeforeUser;
    }
}

File: app/src/main/java/me/echeung/moemoekyun/client/socket/response/UpdateResponse.kt
package me.echeung.moemoekyun.client.socket.response

import me.echeung.moemoekyun.client.model.Event
import me.echeung.moemoekyun.client.model.Song
import me.echeung.moemoekyun.client.model.User

class UpdateResponse : BaseResponse() {
    val t: String? = null
    val d: Details? = null

    class Details {
        val song: Song? = null
        val startTime: String? = null
        val lastPlayed: List<Song>? = null
        val queue: Queue? = null
        val listeners: Int = 0
        val requester: User? = null
        val event: Event? = null
    }

    class Queue {
        val inQueue: Int = 0
        val inQueueByUser: Int = 0
        val inQueueBeforeUser: Int = 0
    }
}
